addons/event_sale/models/sale_order.py

```diff
@@ -8,6 +8,7 @@ class SaleOrder(models.Model):
     _inherit = "sale.order"
 
     attendee_count = fields.Integer('Attendee Count', compute='_compute_attendee_count')
+    event_registration_ids = fields.One2many('event.registration', 'sale_order_id', readonly=True)
```

Comment:
Seems only used for the confirmation page so I think we could avoid this field by getting the registrations from the sale order in `_prepare_shop_payment_confirmation_values`.
With something like this:
`values['attendees'] = request.env['event.registration'].search([('sale_order_id', '=', order.id)])`